diff --git a/settingsSection.tsx b/settingsSection.tsx
index 8cad912be88c51dca4901f094c3939b7e354a98b..94ac6cc39a7bba8c71a412134bd6cde749bc2b14 100644
--- a/settingsSection.tsx
+++ b/settingsSection.tsx
@@ -8,18 +8,20 @@ import {
   ISettingsFieldToggle,
   NewValueTypes,
 } from "./types/settings-field";
+const ButtonSecondary = Spicetify.ReactComponent.ButtonSecondary;
 
 class SettingsSection {
-  settingsFields: { [nameId: string]: ISettingsField } =
-    this.initialSettingsFields;
+  settingsFields: { [nameId: string]: ISettingsField } = {};
   private stopHistoryListener: any;
   private setRerender: Function | null = null;
 
   constructor(
-    public name: string,
+    public title: string,
     public settingsId: string,
-    public initialSettingsFields: { [key: string]: ISettingsField } = {}
-  ) {}
+    initialSettingsFields: { [nameId: string]: ISettingsField } = {}
+  ) {
+    this.settingsFields = initialSettingsFields;
+  }
 
   pushSettings = async () => {
     Object.entries(this.settingsFields).forEach(([nameId, field]) => {
@@ -101,20 +103,15 @@ class SettingsSection {
   addInput = (
     nameId: string,
     description: string,
-    defaultValue: string,
-    onChange?: () => void,
-    inputType?: string,
-    events?: ISettingsFieldInput["events"]
+    opts: Omit<ISettingsFieldInput, "type" | "description" | "defaultValue"> & {
+      defaultValue?: string;
+    } = {}
   ) => {
     this.settingsFields[nameId] = {
       type: "input",
-      description: description,
-      defaultValue: defaultValue,
-      inputType: inputType,
-      events: {
-        onChange: onChange,
-        ...events,
-      },
+      description,
+      defaultValue: "",
+      ...opts,
     };
   };
 
@@ -165,13 +162,13 @@ class SettingsSection {
 
   getFieldValue = <Type,>(nameId: string): Type => {
     return JSON.parse(
-      Spicetify.LocalStorage.get(`${this.settingsId}.${nameId}`) || "{}"
+      Spicetify.LocalStorage.get(`spicetify:${this.settingsId}:${nameId}`) || "{}"
     )?.value;
   };
 
   setFieldValue = (nameId: string, newValue: any) => {
     Spicetify.LocalStorage.set(
-      `${this.settingsId}.${nameId}`,
+      `spicetify:${this.settingsId}:${nameId}`,
       JSON.stringify({ value: newValue })
     );
   };
@@ -182,8 +179,51 @@ class SettingsSection {
 
     return (
       <div className="x-settings-section" key={rerender}>
+        <style>
+          {`
+            .x-settings-input {
+              letter-spacing: .24px;
+              color: var(--text-base);
+              background-color: var(--background-elevated-base);
+              border: 0;
+              border-radius: 4px;
+              width: 100%;
+              height: 32px;
+              padding-inline: 12px;
+              font-family: var(--encore-body-font-stack, var(--fallback-fonts, sans-serif)), sans-serif;
+              font-size: 14px;
+              font-weight: 400;
+              line-height: 20px;
+              box-sizing: border-box;
+              outline: none;
+            }
+
+            .x-settings-input[type="number"] {
+              appearance: textfield;
+            }
+
+            .x-settings-input[type="number"]::-webkit-outer-spin-button,
+            .x-settings-input[type="number"]::-webkit-inner-spin-button {
+              -webkit-appearance: none;
+              margin: 0;
+            }
+
+            .x-settings-input::placeholder {
+              color: var(--text-subdued);
+            }
+
+            .x-settings-input:hover {
+              background-color: var(--background-elevated-highlight);
+            }
+
+            .x-settings-input:focus {
+              outline: var(--encore-focus-outline-width, var(--encore-border-width-focus)) solid var(--text-base);
+              outline-offset: 0;
+            }
+          `}
+        </style>
         <h2 className="TypeElement-cello-textBase-type">
-          {this.name}
+          {this.title}
         </h2>
         {Object.entries(this.settingsFields).map(([nameId, field]) => {
           return <this.Field nameId={nameId} field={field} />;
@@ -208,7 +248,7 @@ class SettingsSection {
 
     const [value, setValueState] = useState(defaultStateValue);
 
-    const setValue = (newValue?: any) => {
+    const setValue = (newValue?: NewValueTypes) => {
       if (newValue !== undefined) {
         setValueState(newValue);
         this.setFieldValue(props.nameId!, newValue);
@@ -227,36 +267,64 @@ class SettingsSection {
         </div>
         <div className="x-settings-secondColumn">
           {props.field.type === "input" ? (
-            <input
-              className="x-settings-input"
-              id={id}
-              dir="ltr"
-              value={value as string}
-              type={props.field.inputType || "text"}
-              {...props.field.events}
-              onChange={(e) => {
-                setValue(e.currentTarget.value);
-                const onChange = (props.field as ISettingsFieldInput).events
-                  ?.onChange;
-                if (onChange) onChange(e);
-              }}
-            />
+            <>
+              <input
+                className="x-settings-input"
+                id={id}
+                dir="ltr"
+                value={value as string}
+                type={props.field.inputType || "text"}
+                list={props.field.suggestions ? `${id}-suggestions` : undefined}
+                {...props.field.props}
+                onChange={(e) => {
+                  setValue(e.currentTarget.value);
+                  const onChange = (props.field as ISettingsFieldInput).props?.onChange;
+
+                  if (onChange) {
+                    const newValue = onChange(e);
+
+                    if (typeof newValue === "string") {
+                      setValue(newValue)
+                    }
+                  }
+                }}
+                onBlur={(e) => {
+                  const onBlur = (props.field as ISettingsFieldInput).props?.onBlur;
+
+                  if (onBlur) {
+                    const newValue = onBlur(e);
+
+                    if (typeof newValue === "string") {
+                      setValue(newValue)
+                    }
+                  }
+                }}
+              />
+
+              {props.field.suggestions && Array.isArray(props.field.suggestions) && (
+                <datalist id={`${id}-suggestions`}>
+                  {props.field.suggestions.map((suggestion, index) => (
+                    <option key={index} value={suggestion} />
+                  ))}
+                </datalist>
+              )}
+            </>
           ) : props.field.type === "button" ? (
             <span>
-              <button
+              <ButtonSecondary
                 id={id}
-                className="Button-sc-y0gtbx-0 Button-small-buttonSecondary-useBrowserDefaultFocusStyle x-settings-button"
+                size="small"
+                className="x-settings-button"
                 {...props.field.events}
-                onClick={(e) => {
+                onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
                   setValue();
                   const onClick = (props.field as ISettingsFieldButton).events
                     ?.onClick;
                   if (onClick) onClick(e);
                 }}
-                type="button"
               >
                 {value}
-              </button>
+              </ButtonSecondary>
             </span>
           ) : props.field.type === "toggle" ? (
             <label className="x-settings-secondColumn x-toggle-wrapper">
diff --git a/types/settings-field.ts b/types/settings-field.ts
index 0b7d15b69de343f182b8f33e618a645d684843b2..aeb58fd617394bc06ce1ae5aaa7fd1e73be01292 100644
--- a/types/settings-field.ts
+++ b/types/settings-field.ts
@@ -6,9 +6,10 @@ export interface ISettingsFieldHidden {
 export interface ISettingsFieldInput {
     type: "input",
     description?: string,
-    defaultValue: any,
-    inputType?: string,
-    events?: Partial<React.InputHTMLAttributes<HTMLInputElement>>,
+    defaultValue: string,
+    suggestions?: string[],
+    inputType?: React.HTMLInputTypeAttribute,
+    props?: Partial<React.InputHTMLAttributes<HTMLInputElement>>,
 }
 
 export interface ISettingsFieldDropdown {
diff --git a/types/spicetify.d.ts b/types/spicetify.d.ts
index fc3ea053ce5c2ea5b30be0efb4ea601bade371b0..d9868262020291703954ae70993529e3f08bd929 100644
--- a/types/spicetify.d.ts
+++ b/types/spicetify.d.ts
@@ -416,12 +416,12 @@ declare namespace Spicetify {
             deregister(): void;
         }
     }
-    
+
     /**
      * Keyboard shortcut library
-     * 
+     *
      * Documentation: https://craig.is/killing/mice v1.6.5
-     * 
+     *
      * Spicetify.Keyboard is wrapper of this library to be compatible with legacy Spotify,
      * so new extension should use this library instead.
      */
@@ -1291,23 +1291,41 @@ declare namespace Spicetify {
              */
             icon?: React.ReactNode;
         };
+        type ButtonSize = 'sm' | 'md' | 'lg';
+        type ButtonProps = Partial<React.ButtonHTMLAttributes<HTMLButtonElement>> & {
+            component?: React.ElementType;
+            buttonSize?: ButtonSize;
+            size?: 'small' | 'medium' | 'large';
+            children?: React.ReactNode;
+            iconLeading?: React.ElementType | string;
+            iconTrailing?: React.ElementType | string;
+            iconOnly?: React.ElementType | string;
+        };
+        type ButtonComponent = React.ForwardRefExoticComponent<
+            ButtonProps & React.RefAttributes<HTMLButtonElement>
+        > & {
+            sm: 'sm';
+            md: 'md';
+            lg: 'lg';
+            displayName?: string;
+        };
         /**
          * Generic context menu provider
-         * 
+         *
          * Props:
          * @see Spicetify.ReactComponent.ContextMenuProps
          */
         const ContextMenu: any;
         /**
          * Wrapper of ReactComponent.ContextMenu with props: action = 'toggle' and trigger = 'right-click'
-         * 
+         *
          * Props:
          * @see Spicetify.ReactComponent.ContextMenuProps
          */
         const RightClickMenu: any;
         /**
          * Outer layer contain ReactComponent.MenuItem(s)
-         * 
+         *
          * Props:
          * @see Spicetify.ReactComponent.MenuProps
          */
@@ -1315,19 +1333,20 @@ declare namespace Spicetify {
         /**
          * Component to construct menu item
          * Used as ReactComponent.Menu children
-         * 
+         *
          * Props:
          * @see Spicetify.ReactComponent.MenuItemProps
          */
         const MenuItem: any;
         /**
          * Tailored ReactComponent.Menu for specific type of object
-         * 
+         *
          * Props: {
          *      uri: string;
          *      onRemoveCallback?: (uri: string) => void;
          * }
          */
+        const ButtonSecondary: ButtonComponent;
         const AlbumMenu: any;
         const PodcastShowMenu: any;
         const ArtistMenu: any;
@@ -1349,7 +1368,7 @@ declare namespace Spicetify {
     }
 
     /**
-     * SVG icons 
+     * SVG icons
      */
     namespace SVGIcons {
         const check: string;
